selfdrive/updated.py

```diff
@@ -303,11 +305,36 @@ def attempt_update():
 
     finalize_from_ovfs_copy()
 
+    # If a NEOS update is required, download it in the background
+    with open("/VERSION", "r") as current_neos_file:
+      current_neos_version = current_neos_file.read().replace("\n", "")
+    required_neos_version = run(["bash", "-c", 
+               r"unset REQUIRED_NEOS_VERSION && source launch_env.sh && echo -n $REQUIRED_NEOS_VERSION"],
+               FINALIZED)
+    print(f"NEOS version update check: {current_neos_version} current, {required_neos_version} in update")
+    if current_neos_version != required_neos_version or DEBUG_FORCE_UPDATE:
+      print(f"Beginning background download for NEOS {required_neos_version}")
+      update_json = f'file:///{FINALIZED}/installer/updater/update.json'
+      while True:
+        Params().put("Offroad_NeosUpdate", "1")
+        try:
+          run(NICE_LOW_PRIORITY + ["installer/updater/updater", "bgcache", update_json], FINALIZED)
+          Params().put("Offroad_NeosUpdate", "0")
+          print("NEOS background download successful!")
+          break
+        except subprocess.CalledProcessError:
+          print("NEOS background download failed, will retry at next wait interval")
+          Params().put("Offroad_NeosUpdate", "0")
```

Comment:
why clear the param? It's still trying to download

Reply:
UX judgment call, could go either way. For not clearing we'll get "why does it show I'm downloading when I'm not even connected to the Internet", for clearing we'll get "why did it say I was getting an update and then disappear". Depends on the wording we want the APK to show the user.